Sources/MongoSwift/Hint.swift

```diff
@@ -17,6 +17,18 @@ public enum IndexHint: Codable {
         }
     }
 
+    public init(stringLiteral value: StringLiteralType) {
+        self = .indexName(value)
+    }
+
+    public init(dictionaryLiteral keyValuePairs: (String, BSON)...) {
+        var docs = BSONDocument()
```

Comment:
So as to not repeat `BSONDocument`'s implementation of this protocol, we can just call straight through to it (e.g. `BSONDocument(dictionaryLiteral: keyValuePairs)`)

Reply:
if I understand correctly I would be doing something like `self = .indexSpec(BSONDocument(dictionaryLiteral: keyValuePairs))` but that gives compiler errors because the types

Reply:
Oh hm, good point. It looks like we'd need https://bugs.swift.org/browse/SR-128 to be implemented before we could do something like that unfortunately. Anyways, sorry for sending you down this dead end, this looks good as-is.